### PR TITLE
Email: Fixed time stamp for event and abbreviation for states

### PR DESCRIPTION
**Pull request by toiletspider**
Fixed time stamp for events and the abbreviation for states.

### References Issues:
Email Ticket Buyers Directly within Studio

### Migrations
 * No Migrations

### Environment Variables
 * No change

### CODE REVIEW

File: api/src/utils/communication/mod.rs [hunk 1]
```diff
@@ -202,3 +219,84 @@ pub fn customer_io_send_email(
     }
     Ok(())
 }
+
+fn convert_state(state: &str) -> Option<String> {
```
Comment (Krakaw) [inline api/src/utils/communication/mod.rs]: https://github.com/JohnBendfeldt/country-province-data/blob/c8db0e2b5bda0404b8f22d2fd08615faa11da6e2/countries.json#L5575-L5575
http://github.com/big-neon/bn-api/blob/5212c66aafa03f4b1e3ef5cbc6706980eeee1097/db/src/services/country_lookup.rs#L203-L203

File: api/src/utils/communication/mod.rs [hunk 2]
```diff
@@ -202,3 +219,23 @@ pub fn customer_io_send_email(
     }
     Ok(())
 }
+
+fn parse_state(state_to_parse: &str) -> String {
+    CountryLookup::new()
+        .ok()
+        .and_then(|c| c.find("US"))
+        .and_then(|countre_dat| countre_dat.convert_state(state_to_parse))
```
Comment (Krakaw) [inline api/src/utils/communication/mod.rs]: What is this variable?

File: api/src/utils/communication/mod.rs [hunk 3]
```diff
@@ -202,3 +219,23 @@ pub fn customer_io_send_email(
     }
     Ok(())
 }
+
+fn parse_state(state_to_parse: &str) -> String {
+    CountryLookup::new()
+        .ok()
+        .and_then(|c| c.find("US"))
```
Comment (mikethetike) [inline api/src/utils/communication/mod.rs]: I think add a check for "US" to cater for non-US countries

File: api/src/utils/communication/mod.rs (hunk 2 above)
Comment (Krakaw) [inline api/src/utils/communication/mod.rs]:
```suggestion
        .and_then(|country_datum| country_datum.convert_state(state_to_parse))
```

**Comment by Krakaw** [comment]
Success!
image

**Pull request merged**